### ISSUE TITLE
Some wonder about function "voxelization" in "if_clight_renderer.py"

### ISSUE THREAD

**Issue by junjxiao** [pansanity666/TransHuman#7]
Hi, thanks for your great work. 

I'm trying to run your code on RTX3090 gpu, however I found it meets "CUDA OUT OF MEMORY" when executing "loss.backward". I think  the reason may bethe accumulated grads, so I add the code "with torch.no_grad()" in function "voxelization", and it works.
Following is the changed code:
```
def voxelization(self, dict_voxel2pc_ind, SMPL, return_list=False):
        # voxelize based on dict

        voxel_list = []
        voxel = []
        with torch.no_grad(): #additional code
            for vn, pc_list in dict_voxel2pc_ind.items():
                # mean pooling 
                voxel.append(SMPL[pc_list].mean(0))
                voxel_list.append(SMPL[pc_list])
        voxel = torch.stack(voxel)  # torch.Size([2115, 67])

        if return_list:
            return voxel, voxel_list
        else:
            return voxel
```

So I wonder will it affect the final performance? Thanks for your reading.

**Comment by junjxiao** [comment]
By the way, would you consider release the code and pretrained model on THUMAN2 dataset? Recently, they have released an extended dataset THUMAN2.1 which includes 2500 3D models. The quality  of ZJU-Mocap is however limited, I think it's a better choice.

**Comment by pansanity666** [comment]
Hi, 

Thanks for your interest. 

1. I believe that stopping the gradient will influence the performance since it stops the gradient to image encoder. However, I haven't tried it. 
2. OOM is mainly caused by the patch sampling for LPIPS training. Please refer to https://github.com/pansanity666/TransHuman/issues/2#issuecomment-1770286181 and use the fast scheduler for saving memory usage. And another option is to reduce the patch size in the config. However, I am not sure both workarounds will reproduce the results. 
3. Currently, I don't have computational resources for running on THUMAN2. Feel free to report it and I will merge the results in README if you have resources. 

Best,
